src/graph/optimizer/rule/PushFilterDownAggregateRule.cpp

```diff
@@ -89,11 +89,13 @@ StatusOr<OptRule::TransformResult> PushFilterDownAggregateRule::transform(
   newFilterNode->setCondition(newCondition);
 
   // Exchange planNode
-  newAggNode->setOutputVar(oldFilterNode->outputVar());
-  newFilterNode->setInputVar(oldAggNode->inputVar());
+  newAggNode->setOutputVar(oldFilterNode->outputVar());  // ~ok
+  newFilterNode->setInputVar(oldAggNode->inputVar());    // ok
   DCHECK_EQ(oldAggNode->outputVar(), oldFilterNode->inputVar());
-  newAggNode->setInputVar(oldAggNode->outputVar());
-  newFilterNode->setOutputVar(oldAggNode->outputVar());
+  // newAggNode->setInputVar(oldAggNode->outputVar());
+  newAggNode->setInputVar(oldAggNode->inputVar());
```

Comment:
I think 
```
newAggNode->setInputVar(newFilterNode->outputVar());
```
 is enough.

Reply:
Agree. But that would be a little bit difficult to understand. Let me add some comments here.

Reply:
`newAggNode->setInputVar(newFilterNode->outputVar());` This seems not ok. I've reverted it. Check the reason later.

Reply:
> I think
> 
> ```
> newAggNode->setInputVar(newFilterNode->outputVar());
> ```
> 
> is enough.

@xtcyclist I also want to know why this fix could not solve the issue. Maybe there exists other bugs if it still reports error.

Reply:
Yes, there is another bug. -1 * -1 = 1. 1 * -1 = -1. Ha, ha, ha.

Reply:
Updated PushFilterDownProjectRule as well to fix this issue. Please check the codes and the added tcks.